backend/Events/LobbyEvents.js

```diff
@@ -0,0 +1,27 @@
+const Player = require("../domain/Player");
+const LobbyCodeDTO = require("../domain/DTO/response/LobbyCodeDTO");
+const MafiaGame = require("../domain/MafiaGame");
+
+/**
+ * Event handlers and logic for `create-lobby` and `lobby-code`
+ * The goal of these lobby events is to allow a host to create a game and receive a new room id. 
+ * @param {any} io 
+ * @param {any} socket 
+ * @param {MafiaGame} mafiaGame 
+ */
+module.exports = function(io, socket, mafiaGame){
+    socket.on("create-lobby", (createLobbyDTO) => {
+        console.log("New room request received");
+        // Create room and assign host player to the room
+        let roomID = mafiaGame.newGame();
+        let host = new Player(socket.id, roomID, createLobbyDTO.nickname);
+        mafiaGame.gameRoomsDict[roomID].addPlayer(host);
+
+        // Subscribe to the room events
+        socket.join(roomID);
```

Comment:
As a quickie, are you able to add the roomID as a property of the socket when creating/joining a lobby? It would make referencing the room of the socket easier for other handlers.

`socket.roomID = roomID;`

Reply:
yeah i think it would be

Reply:
sorry, had a discussion with the team and we decided to obtain the room through this method.

https://stackoverflow.com/questions/19044660/socket-io-get-rooms-which-socket-is-currently-in

it would be best to revert the change